src/m365/spp/commands/contentcenter/contentcenter-list.ts

```diff
@@ -0,0 +1,74 @@
+import { Logger } from '../../../../cli/Logger.js';
+import config from '../../../../config.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { ClientSvcResponse, ClientSvcResponseContents, FormDigestInfo, spo } from '../../../../utils/spo.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import { SiteProperties } from '../../../spo/commands/site/SiteProperties.js';
+import { SPOSitePropertiesEnumerable } from '../../../spo/commands/site/SPOSitePropertiesEnumerable.js';
+import commands from '../../commands.js';
+
+class SppContentCenterListCommand extends SpoCommand {
+  private allSites?: SiteProperties[];
+
+  public get name(): string {
+    return commands.CONTENTCENTER_LIST;
+  }
+
+  public get description(): string {
+    return 'Gets the URLs of the SharePoint Premium content centers';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['Title', 'Url'];
+  }
+
+  public async commandAction(logger: Logger): Promise<void> {
+    try {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving list of content centers...`);
+      }
+
+      this.allSites = [];
+
+      const spoAdminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
+      await this.getAllSites(spoAdminUrl, '0', logger);
+      await logger.log(this.allSites);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getAllSites(spoAdminUrl: string, startIndex: string | undefined, logger: Logger): Promise<void> {
+    const res: FormDigestInfo = await spo.ensureFormDigest(spoAdminUrl, logger, undefined, this.debug);
```

Comment:
Let's rework this so that we use a 
do {

} while (sites.NextStartIndexFromSharePoint);

Reply:
done